cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/passes/VariableUsageResolver.kt

```diff
@@ -118,9 +118,21 @@ open class VariableUsageResolver : SymbolResolverPass() {
 
         // For now, we need to ignore reference expressions that are directly embedded into call
         // expressions, because they are the "callee" property. In the future, we will use this
-        // property to actually resolve the function call.
+        // property to actually resolve the function call. However, there is a special case that
+        // we want to catch already, that is if we are "calling" a reference to a variable. This
+        // can be done in several languages, e.g., in C/C++ as function pointers or in Go as
+        // function references. In this case, we want to resolve the declared reference expression
+        // of this call expression back to its original variable declaration. In the future, we want
+        // to extend this particular code to resolve all callee references to their declarations,
+        // i.e., their function definitions and get rid of the separate CallResolver.
         if (parent is CallExpression && parent.callee === current) {
-            return
+            // Peek into the declaration, and if it is a variable, we can proceed normally, as we
+            // are running into the special case explained above. Otherwise, we abort here (for
+            // now).
+            val wouldResolveTo = scopeManager.resolveReference(current, current.scope)
```

Comment:
Unfortunately, this now calls `resolveReference` twice in the case of a call expression reference. Anyone got a good idea how to merge this with line 139?

Reply:
Maybe not a good idea but you could make a `var wouldResolveTo: Declaration? = nukk` before line 128, set it in line 132, and only call `resolveReference` in line 139 if it is null

Reply:
Its better than my solution, so sold!